synapse/handlers/federation.py

```diff
@@ -746,7 +746,6 @@ async def on_make_join_request(
 
         # The remote hasn't signed it yet, obviously. We'll do the full checks
         # when we get the event back in `on_send_join_request`
-        validate_event_for_room_version(room_version, event, do_sig_check=False)
```

Comment:
this means we're not checking that the event has a `room_id` attribute, or the sizes of the event fields.

The former seems fine, given we've just built the event. The latter maybe less so, but it'll be checked multiple other times before it gets into the event DAG, so I'm going to go with it anyway.

Reply:
> this means we're not checking that the event has a `room_id` attribute

The `room_id` check seems to have been moved into `check_auth_rules_for_event` so I think that will still get called?

Reply:
I agree that skipping checking the size of the fields isn't a huge deal here since we're not persisting anything into the room until the `send_*` API call comes back.

Reply:
> The `room_id` check seems to have been moved into `check_auth_rules_for_event` so I think that will still get called?

I don't think so? I'm referring to this check: https://github.com/matrix-org/synapse/pull/10940/files#diff-2f0812c5ff333e8ee94dd38210fc77f98ccd0d8a995f9864d8c839e311b3f1a9R72-R73

Reply:
Right, we do still pull out the `room_id` though so I'd expect it to error at least: https://github.com/matrix-org/synapse/pull/10940/files#diff-2f0812c5ff333e8ee94dd38210fc77f98ccd0d8a995f9864d8c839e311b3f1a9R149

Anyway, I think it is fine.